### PR TITLE
[Test] Agent leak release test (stress test)

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This PR adds a release regression test for dashboard agent memory leak. 

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/dashboard/agent_stress_app_config.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+base_image: {{ env["RAY_IMAGE_NIGHTLY_GPU"] | default("anyscale/ray-ml:2.0.0-py37-gpu") }}
+debian_packages: []
+env_vars: {"RAY_INTERNAL_MEM_PROFILE_COMPONENTS": "dashboard_agent"}
+debian_packages:
+  - htop
+  - curl
+
+python:
+  pip_packages:
```
Comment (rickyyx) [inline release/dashboard/agent_stress_app_config.yaml:9]: Are all these needed?

File: release/dashboard/atari.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+atari-ppo:
```
Comment (rickyyx) [inline release/dashboard/atari.yaml:1]: is this based on some other release test?

File: release/release_tests.yaml [hunk 3]
```diff
@@ -4747,6 +4747,28 @@
...
+  stable: false
+
+  frequency: nightly
+  team: core
+  cluster:
+    cluster_env: agent_stress_app_config.yaml
+    cluster_compute: agent_stress_compute.yaml
+
+  run:
+    timeout: 14400
```
Comment (rickyyx) [inline release/release_tests.yaml:4766]: this gonna be 4 hrs right? Is this enough?

**Review by rickyyx — DISMISSED** [review]
Great test! The test logic and code LGTM. We should definitely do this. 

A couple of questions (so that we have them documented when we adding new release test) just for iteration purpose. 

- How much will this cost?
- Why are we using the rlib workload given we have long running tests, could we piggyback memory leak check on those long running tests? 


cc @scv119 

Let's kick off one btw to see how this goes.

**Comment by rkooo567** [comment]
> How much will this cost?
This is cheap. Just one m5.16xlarge * 4 hours ($12). 

> Why are we using the rlib workload given we have long running tests, could we piggyback memory leak check on those long running tests?

It will be easy to adjust load & num workers & other config this way. I think that's important for release tests and also debugging.

**Comment by rkooo567** [comment]
Btw re; memory leak check. I think we should definitely do it in all tests. What I hope is to have some sort of alert to prometheus, but I am not sure how easy it is to implement it in prod...

File: release/dashboard/atari.yaml (hunk 2 above)
Comment (rkooo567) [reply release/dashboard/atari.yaml:1]: no. This is the repro script for the memory leak.

File: release/release_tests.yaml (hunk 3 above)
Comment (rkooo567) [reply release/release_tests.yaml:4766]: I think 4h is enough to see if there's bad memory leak

File: release/dashboard/agent_stress_app_config.yaml (hunk 1 above)
Comment (rkooo567) [reply release/dashboard/agent_stress_app_config.yaml:9]: yes!

**Comment by rkooo567** [comment]
Running test now

**Comment by rkooo567** [comment]
For some reasons, release test cannot find the wheel from the this PR. Let's merge it and see how it works (it worked when I ran from my laptop). It is an unstable test, so the impact will be low

**Review by scv119 — APPROVED** [review]

**Pull request merged**